### PR TITLE
add `Rewrite with new syntax` code lens on top level classes and code actions.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
1. add `Rewrite with new syntax` code lens on top level classes and code actions to trigger inspecting classes.

### CODE REVIEW

File: src/copilot/inspect/Inspection.ts [hunk 1]
```diff
@@ -31,11 +31,30 @@ export interface Inspection {
 }
 
 export namespace Inspection {
-    export function fix(inspection: Inspection, source: string) {
-        //TODO: implement me
+    export function selectFirstLineOfInspection(inspection: Inspection) {
```
Comment (testforstephen) [inline src/copilot/inspect/Inspection.ts:34]: -> revealFirstLineOfInspection ?

File: src/copilot/inspect/commands.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        const copilot = new InspectionCopilot();
+        void copilot.inspectClass(document, clazz);
+    });
+
+    instrumentOperationAsVsCodeCommand(COMMAND_INSPECT_RANGE, async (document: TextDocument, range: Range | Selection) => {
+        const copilot = new InspectionCopilot();
+        void copilot.inspectRange(document, range);
+    });
+
+    instrumentOperationAsVsCodeCommand(COMMAND_FIX, async (problem: Inspection['problem'], solution: string, source) => {
```
Comment (testforstephen) [inline src/copilot/inspect/commands.ts:22]: add the type for `source` argument.

File: src/copilot/inspect/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+export async function activateCopilotInspection(context: ExtensionContext): Promise<void> {
+
+    registerCommands();
+
+    const inspectActionCodeLenses = new InspectActionCodeLensProvider().install(context);
+
+    context.subscriptions.push(
+        workspace.onDidOpenTextDocument(doc => inspectActionCodeLenses.rerender(doc)), // Rerender class codelens when open a new document
+        workspace.onDidChangeTextDocument(e => inspectActionCodeLenses.rerender(e.document)), // Rerender class codelens when change a document
```
Comment (testforstephen) [inline src/copilot/inspect/index.ts:15]: do you think it's necessary to debounce the onDidChangeTextDocument event?

File: src/copilot/inspect/Inspection.ts [hunk 4]
```diff
@@ -31,11 +31,30 @@ export interface Inspection {
...
+                const range = document.lineAt(inspection.problem.position.line).range;
+                editor.selection = new Selection(range.start, range.end);
+                editor.revealRange(range);
+            });
+        });
     }
 
-    export function highlight(inspection: Inspection) {
-        //TODO: implement me
+    export function getIndicatorRangeOfInspection(problem: Inspection['problem']): Range {
```
[the remark this answers is not in the document]
Comment (testforstephen) [reply src/copilot/inspect/Inspection.ts:44]: Not quite understand this function is for?

**Review by testforstephen — CHANGES_REQUESTED** [review]

File: src/copilot/inspect/InspectActionCodeLensProvider.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        })));
+        this.inspectCodeLenses.set(document.uri, docCodeLenses);
+        this.refresh();
+    }
+
+    public provideCodeLenses(document: TextDocument): CodeLens[] {
+        return this.inspectCodeLenses.get(document.uri) ?? [];
+    }
+
+    public refresh(): void {
```
[the remark this answers is not in the document]
Comment (wangmingliang-ms) [reply src/copilot/inspect/InspectActionCodeLensProvider.ts:36]: method is inlined.

File: src/copilot/inspect/Inspection.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (wangmingliang-ms) [reply src/copilot/inspect/Inspection.ts:48]: this method get the range of the indicator of given inspection.
`indicator` will be used as the position of code lens/diagnostics and also used as initial selection for fix commands.

File: src/copilot/inspect/index.ts (hunk 3 above)
Comment (wangmingliang-ms) [reply src/copilot/inspect/index.ts:15]: good point, it's debounced in later PRs.

**Review by testforstephen — APPROVED** [review]

**Pull request merged**